src/neuroglancer/chunk_manager/backend.ts

```diff
@@ -649,15 +760,15 @@ export class ChunkQueueManager extends SharedObjectCounterpart {
       // Note: After calling this, chunk may no longer be valid.
       this.updateChunkState(chunk, ChunkState.QUEUED);
     };
-    let promotionCandidates = this.queuedPromotionQueue.candidates();
+    let downloadPromotionCandidates = this.queuedDownloadPromotionQueue.candidates();
     let downloadEvictionCandidates = this.downloadEvictionQueue.candidates();
     let systemMemoryEvictionCandidates = this.systemMemoryEvictionQueue.candidates();
     let downloadCapacity = this.downloadCapacity;
     let systemMemoryCapacity = this.systemMemoryCapacity;
     while (true) {
```

Comment:
It would be nice to factor out this logic so that it doesn't need to be repeated for the download and compute queues.